src/main.rs

```diff
@@ -146,29 +149,33 @@ fn main() {
     #[cfg(not(feature = "cli"))]
     compile_error!("The 'cli' feature is needed for the hvm-lang cli");
 
-    let cli = Cli::parse();
+    let mut cli = Cli::parse();
     let arg_verbose = cli.verbose;
+    let entrypoint = cli.entrypoint.take();
+
+    let book = |path: &Path| {
```

Comment:
If this is the only place we do this, why not just put it inside `execute_cli_mode` instead of passing a function?